Add Release Notes for v2.15.0 (#1263)
rancher/rancher-product-docs: versions/v2.15/modules/en/pages/release-notes/v2.15.0.adoc

* Added the `hide-local-auth-provider` feature flag (disabled by default) to restrict local authentication usage once an External Authentication Provider (EAP) is active. When enabled, the flag hides the local login interface and removes the ability to create, edit, or delete local users from the UI. Note that password rotation for required administrative local accounts remains allowed to support emergency fallback workflows. See https://github.com/rancher/rancher/issues/46078[#46078].

[#_cluster_provisioning]
== Cluster Provisioning

[#_features_and_enhancements_3]
=== Features and Enhancements

* Rancher v2.15 adds support for Cluster API (CAPI) v1.13.2. See https://github.com/rancher/rancher/issues/54304[#54304].
* Users with the `clusters-create` role can now view the CAPIprovider resource. See https://github.com/rancher/rancher/issues/55167[#55167].

[#_k3s_and_rke2_provisioning]
== K3s and RKE2 Provisioning

[#_features_and_enhancements_4]
=== Features and Enhancements

> * **General Availability**: Promoted support for native Cluster API (CAPI) infrastructure providers within Rancher’s V2 provisioning framework (v2prov) from Tech Preview to General Availability (GA). This provides full production support and enhanced reliability when using native CAPI infrastructure providers (e.g., CAPA, CAPZ, CAPG) for cluster lifecycle management. See https://github.com/rancher/rancher/issues/53777[#53777].

Reviewer: ```suggestion
* **General Availability**: Promoted support for native Cluster API (CAPI) infrastructure providers within Rancher’s V2 provisioning framework (v2prov) from Tech Preview to General Availability (GA). This provides full production support and enhanced reliability when using native CAPI infrastructure providers (e.g., CAPA and CAPV) for cluster lifecycle management. See https://github.com/rancher/rancher/issues/53777[#53777].
```

Author: Hi @pedromfcarvalho thanks for reviewing, added this suggestion and the one below in this [ref commit](https://github.com/rancher/rancher-product-docs/pull/1263/commits/dd3e9e3b4a27b2a6cb79a9f736d67bc97279d742).

[#_known_issues]
=== Known Issues

> * A known issue exists related to CAPI infrastructure providers within Rancher’s V2 provisioning framework. Editing machine pools or experiencing provisioning failures on CAPI Plug-and-Play (PnP) clusters (such as CAPA/AWS) can leave unused, orphaned infrastructure templates (e.g., `AWSMachineTemplate` or `AWSCluster`) in the management cluster. These dangling resources are no longer referenced by active `MachineDeployments` and can be safely cleaned up manually if needed. See https://github.com/rancher/rancher/issues/55752[#55752].

Reviewer: ```suggestion
* A known issue exists related to CAPI infrastructure providers within Rancher’s V2 provisioning framework. Creating or editing v2prov clusters with native infrastructure providers (such as CAPA or CAPV) from the UI can under some failure conditions leave unused, orphaned infrastructure resources (e.g., `AWSMachineTemplate` or `AWSCluster`) in the management cluster which need to be cleaned up manually. See https://github.com/rancher/rancher/issues/55752[#55752].
```

[#_behavior_changes_rancher_general]
=== Behavior Changes

* Rancher v2.15 removes support for Kubernetes v1.33. See https://github.com/rancher/rancher/issues/55306[#55306].

[#_retention_policy_for_rancher_app_charts]
==== Retention Policy for Rancher App Charts

To improve repository performance, Rancher is introducing a lifecycle management policy for charts available in the Apps feature of Rancher, specifically in the "Rancher" repository.

* *The Policy:* Rancher will transition from a cumulative model (retaining all historical versions forever) to a retention model that preserves chart versions for the *seven (7) most recent Rancher minor releases* (approximately a 2.5-year window).
* *Timeline - Rancher v2.13 & v2.14:* Legacy chart versions (older than the 7-version window) remain available.
* *Rancher v2.15:* This release is the first version to enforce the policy. Versions falling outside the 7-version window and older than two years will no longer be available.

> *Impact:* This change is non-destructive for existing Rancher installations. Historical versions will remain accessible but will not be available in newer release branches once they age out of the 7-version window. You are advised to upgrade your applications before upgrading to Rancher v2.15. Uninstallation after v2.15, and replacement with an updated version, will still be possible.

Reviewer 2: ```suggestion
*Impact:* This change is non-destructive for existing Rancher installations. Historical versions will remain accessible but will not be available in newer release branches once they age out of the 7-version window. You are advised to upgrade your applications before upgrading to Rancher v2.15. Uninstallation after v2.15 and replacement with an updated version will still be possible.
```

Author: Thanks @LucasSaintarbor updated with your suggestions in this [commit](https://github.com/rancher/rancher-product-docs/pull/1263/commits/c1893a018189086d98ba07cd4df8d8e0e4390773).

[#_rancher_app_global_ui]
== Rancher App (Global UI)

[#_features_and_enhancements_rancher_app_global_ui]
=== Features and Enhancements

* The Rancher Dashboard replaces the legacy all-in-one Workloads list with a new **Workloads Overview** UI view to solve major performance and scalability issues when viewing workload resources. See https://github.com/rancher/dashboard/issues/11513[#11513].

[#_behavior_changes_rancher_app_global_ui]
=== Behavior Changes

* The option to disable the feature flag `ui-sql-cache` / Vai / Server-Side Pagination (SSP) in the Rancher UI has been removed in preparation for Rancher v2.16, where the feature will always be enabled. See https://github.com/rancher/dashboard/issues/16822[#16822].
* **Removal**: Support for Ember-based UI plugins for cluster and node drivers (deprecated since Rancher v2.11.0) is removed in Rancher v2.15.0. Migrate your plugins to the https://extensions.rancher.io/[UI Extensions Framework]. For details, see https://github.com/rancher/dashboard/issues/14005[#14005].

[#_major_bug_fixes_rancher_app_global_ui]
=== Major Bug Fixes

> * Fixed an issue where selecting *Enable Dual-Stack* with *Stack Preference* set to *IPv6* omitted the required UI warning. Dual-stack clusters require the *Stack Preference* option to be set to *Dual*. For details, see https://github.com/rancher/dashboard/issues/16788[issue #16788].

Reviewer 2: ```suggestion
* Fixed an issue where selecting *Enable Dual-Stack* with *Stack Preference* set to *IPv6* omitted the required UI warning. Dual-stack clusters require the *Stack Preference* option to be set to *Dual*. For details, see https://github.com/rancher/dashboard/issues/16788[#16788].
```

[#_authentication]
== Authentication

[#_features_and_enhancements_authentication]
=== Features and Enhancements

> * Azure AD SSO Logout: Rancher now supports Single Logout (SLO) for the Azure AD auth provider. When enabled, logging out of Rancher also terminates the user's Azure AD session by redirecting to Azure AD's end-session endpoint. A `LogoutEndpoint` override field is available for sovereign-cloud deployments (Azure US Government, China). An optional `LogoutAllForced` flag can prevent users from bypassing SSO logout. See https://github.com/rancher/rancher/issues/54566[#54566].

Reviewer 2: ```suggestion
* *Azure AD SSO Logout*: Rancher now supports Single Logout (SLO) for the Azure AD auth provider. When enabled, logging out of Rancher also terminates the user's Azure AD session by redirecting to Azure AD's end-session endpoint. A `LogoutEndpoint` override field is available for sovereign-cloud deployments (Azure US Government, China). An optional `LogoutAllForced` flag can prevent users from bypassing SSO logout. See https://github.com/rancher/rancher/issues/54566[#54566].
```

* *Requirements for air-gapped environments:*
** When using a proxy in front of an air-gapped Rancher instance, you must pass additional parameters to `NO_PROXY`. See the {rn-install-rancher}[documentation] and issue https://github.com/rancher/docs/issues/2725#issuecomment-702454584[#2725].
** When installing Rancher with Docker in an air-gapped environment, you must supply a custom `registries.yaml` file to the `docker run` command, as shown in the {rn-private-registry}[K3s documentation]. If the registry has certificates, then you'll also need to supply those. See https://github.com/rancher/rancher/issues/28969#issuecomment-694474229[#28969].

[#_versions]
== Versions

[#_images]
=== Images

* rancher/rancher:{release-version}

[#_tools]
=== Tools

* CLI - https://github.com/rancher/cli/releases/tag/{release-version}[{release-version}]

[#_kubernetes_versions_for_rke2_k3s]
=== Kubernetes Versions for RKE2/K3s

* v1.36.1 (Default)

> * v1.35.6

Reviewer 3: https://github.com/rancher/rancher/issues/56268

```suggestion
* v1.36.3 (Default)
* v1.35.7
```

Author: Thanks @btat update here https://github.com/rancher/rancher-product-docs/pull/1263/commits/9b6aaf2b2a1070c70af35fe83875e3698144f52d.

:rn-back-up-rancher: xref:rancher-admin/back-up-restore-and-disaster-recovery/back-up.adoc
:rn-install-rancher: xref:installation-and-upgrade/other-installation-methods/http-proxy/install-rancher.adoc
endif::[]

[IMPORTANT]
====
* If you are using Active Directory Federation Service (AD FS), upgrading to Rancher v2.10.1 or later may cause issues with authentication, requiring manual intervention. These issues are due to the AD FS Relying Party Trust not being able to pick up a signature verification certificate from the metadata. For more information, see https://github.com/rancher/rancher/issues/48655[#48655]. These issues can be corrected by either of two methods:
** Updating the Relying Party Trust information from federation metadata (Relying Party Trust -> Update from Federation Metadata...).
** Directly adding the certificate (Relying Party Trust -> Properties -> Signature tab -> Add -> Select the certificate).
====

ifeval::["{build-type}" == "community"]
Rancher {release-version} is the latest minor release of Rancher. This is a Community version release that introduces new features, enhancements, and various updates.
endif::[]

[#_rancher_general]
== Rancher General

> [#_features_and_enhancements]

Reviewer 3: ```suggestion
[#_features_and_enhancements_rancher_general]
```

Author: General update to headers in this [commit](https://github.com/rancher/rancher-product-docs/pull/1263/commits/1793db605122af322d5f379ca1a59b3b0c6200dd).

=== Features and Enhancements

* Rancher now supports Kubernetes v1.36. See https://github.com/rancher/rancher/issues/54303[#54303] for information on Rancher support for Kubernetes v1.36. You can view the https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.36.md[upstream Kubernetes changelog] for v1.36 for a complete list of changes.
* Added a cap on the maximum number of public endpoint annotations (`field.cattle.io/publicEndpoints`) written for Ingress objects. This prevents workloads with extensive custom domain lists from exceeding Kubernetes’ 256 KB metadata annotation size limit and triggering API validation errors. See https://github.com/rancher/rancher/issues/42058[#42058].

> * Optimized `rancher-images.txt` to include only the latest patch release per Kubernetes minor version compatible with the target Rancher release. RC pre-releases and superseded patch/minor versions are now filtered out, significantly reducing the size of air-gapped image tarballs while ensuring `system-agent-installer` and auxiliary images remain fully up to date. See https://github.com/rancher/rancher/issues/46174[#46174].

Reviewer 2: ```suggestion
* Optimized `rancher-images.txt` to include only the latest patch release per Kubernetes minor version compatible with the target Rancher release. Release Candidate (RC) pre-releases and superseded patch/minor versions are now filtered out, significantly reducing the size of air-gapped image tarballs while ensuring `system-agent-installer` and auxiliary images remain fully up to date. See https://github.com/rancher/rancher/issues/46174[#46174].
```

> [#_behavior_changes]

Reviewer 3: ```suggestion
[#_behavior_changes_rancher_general]
```

=== Behavior Changes

> * Rancher v2.15 removes support for Kubernetes 1.33. See https://github.com/rancher/rancher/issues/55306[#55306].

Reviewer 3: The entry above about adding 1.36 uses a `v`.

```suggestion
* Rancher v2.15 removes support for Kubernetes v1.33. See https://github.com/rancher/rancher/issues/55306[#55306].
```

> [#_major_bug_fixes]

Reviewer 3: ```suggestion
[#_major_bug_fixes_rancher_general]
```

=== Major Bug Fixes

* Fixed an issue where `rancher-save-images.sh` failed to pull OCI-based Helm charts when the local Docker daemon was configured to use the `overlay2` storage driver. See https://github.com/rancher/rancher/issues/55068[#55068].

[#_rancher_app_global_ui]
== Rancher App (Global UI)

> [#_features_and_enhancements_1]

Reviewer 3: ```suggestion
[#_features_and_enhancements_rancher_app_global_ui]
```

=== Features and Enhancements

* The Rancher Dashboard replaces the legacy all-in-one Workloads list with a new **Workloads Overview** UI view to solve major performance and scalability issues when viewing workload resources. See https://github.com/rancher/dashboard/issues/11513[#11513].

> [#_behavior_changes_1]

Reviewer 3: ```suggestion
[#_behavior_changes_rancher_app_global_ui]
```

=== Behavior Changes

> * The option to disable the feature flag `ui-sql-cache` / Vai / Server-Side Pagination (SSP) in the Rancher UI has been removed in preparation for Rancher v2.16, where the feature will always be enabled. See https://github.com/rancher/dashboard/issues/16822[#16822].

Reviewer 2: ```suggestion
* The option to disable the feature flag `ui-sql-cache` / Vai / Server-Side Pagination (SSP) in the Rancher UI has been removed in preparation for Rancher v2.16, where the feature will be enabled by default. See https://github.com/rancher/dashboard/issues/16822[#16822].
```

> * **Removal**: Support for Ember-based UI plugins for cluster and node drivers (deprecated since Rancher v2.11.0) is removed in Rancher v2.15.0. Migrate your plugins to the https://extensions.rancher.io/[UI Extensions Framework]. For details, see https://github.com/rancher/dashboard/issues/14005[issue #14005].

Reviewer 3: ```suggestion
* **Removal**: Support for Ember-based UI plugins for cluster and node drivers (deprecated since Rancher v2.11.0) is removed in Rancher v2.15.0. Migrate your plugins to the https://extensions.rancher.io/[UI Extensions Framework]. For details, see https://github.com/rancher/dashboard/issues/14005[#14005].
```

Author: Updated this and above suggestion [here](https://github.com/rancher/rancher-product-docs/pull/1263/commits/f7090360b496d4997915094fab2ee2120cd6d122).

> [#_major_bug_fixes_1]

Reviewer 3: ```suggestion
[#_major_bug_fixes_rancher_app_global_ui]
```

> * **Removal**: Support for UI Plugins based on Ember, deprecated since Rancher v2.11.0, regarding cluster and node drivers are **removed** in Rancher v2.15.0. If you still need these plugins, they will need to be migrated to the https://extensions.rancher.io/extensions/next/home[UI Extensions framework]. See https://github.com/rancher/dashboard/issues/14005[#14005] for more information.

Reviewer 2: ```suggestion
* **Removal**: Support for Ember-based UI plugins for cluster and node drivers (deprecated since Rancher v2.11.0) is removed in Rancher v2.15.0. Migrate your plugins to the https://extensions.rancher.io/[UI Extensions Framework]. For details, see https://github.com/rancher/dashboard/issues/14005[issue #14005].
```

=== Major Bug Fixes

* Fixed an issue where selecting *Enable Dual-Stack* with *Stack Preference* set to *IPv6* omitted the required UI warning. Dual-stack clusters require the *Stack Preference* option to be set to *Dual*. For details, see https://github.com/rancher/dashboard/issues/16788[issue #16788].

[#_authentication]
== Authentication

> [#_features_and_enhancements_2]

Reviewer 3: ```suggestion
[#_features_and_enhancements_authentication]
```

> * Fixed an issue when Stack Preference was set to `IPv6` and **Enable Dual-Stack** was checked, the UI modal would not include the Stack Preference warning, giving users the impression that `IPv6` is a valid stack preference for a Dual-Stack cluster when they must set it as `Dual`. See https://github.com/rancher/dashboard/issues/16788[#16788].

Reviewer 2: ```suggestion
* Fixed an issue where selecting *Enable Dual-Stack* with *Stack Preference* set to *IPv6* omitted the required UI warning. Dual-stack clusters require the *Stack Preference* option to be set to *Dual*. For details, see https://github.com/rancher/dashboard/issues/16788[issue #16788].
```

=== Features and Enhancements

* Azure AD SSO Logout: Rancher now supports Single Logout (SLO) for the Azure AD auth provider. When enabled, logging out of Rancher also terminates the user's Azure AD session by redirecting to Azure AD's end-session endpoint. A `LogoutEndpoint` override field is available for sovereign-cloud deployments (Azure US Government, China). An optional `LogoutAllForced` flag can prevent users from bypassing SSO logout. See https://github.com/rancher/rancher/issues/54566[#54566].
* Added the `hide-local-auth-provider` feature flag (disabled by default) to restrict local authentication usage once an External Authentication Provider (EAP) is active. When enabled, the flag hides the local login interface and removes the ability to create, edit, or delete local users from the UI. Note that password rotation for required administrative local accounts remains allowed to support emergency fallback workflows. See https://github.com/rancher/rancher/issues/46078[#46078].

[#_cluster_provisioning]
== Cluster Provisioning

> [#_features_and_enhancements_3]

Reviewer 3: ```suggestion
[#_features_and_enhancements_cluster_provisioning]
```

=== Features and Enhancements

* Rancher v2.15 adds support for Cluster API (CAPI) v1.13.2. See https://github.com/rancher/rancher/issues/54304[#54304].
* Users with the `clusters-create` role can now view the CAPIprovider resource. See https://github.com/rancher/rancher/issues/55167[#55167].

[#_k3s_and_rke2_provisioning]
== K3s and RKE2 Provisioning

> [#_features_and_enhancements_4]

Reviewer 3: ```suggestion
[#_features_and_enhancements_k3s_and_rke2_provisioning]
```

=== Features and Enhancements

* **General Availability**: Promoted support for native Cluster API (CAPI) infrastructure providers within Rancher’s V2 provisioning framework (v2prov) from Tech Preview to General Availability (GA). See https://github.com/rancher/rancher/issues/53777[#53777].
** If the CAPA infrastructure provider is installed, Rancher will mirror {rn-cloud-credentials}[cloud credentials] for AWS and create a corresponding `AWSClusterStaticIdentity` object that targets the `fleet-default` namespace.

> [#_behavior_changes_2]

Reviewer 3: ```suggestion
[#_behavior_changes_k3s_and_rke2_provisioning]
```

> * **General Availability**: Promoted support for native Cluster API (CAPI) infrastructure providers within Rancher’s V2 provisioning framework (v2prov) from Tech Preview to General Availability (GA). This provides full production support and enhanced reliability when using native CAPI infrastructure providers (e.g., CAPA and CAPV) for cluster lifecycle management. See https://github.com/rancher/rancher/issues/53777[#53777].

Reviewer 4: ```suggestion
* **General Availability**: Promoted support for native Cluster API (CAPI) infrastructure providers within Rancher’s V2 provisioning framework (v2prov) from Tech Preview to General Availability (GA). See https://github.com/rancher/rancher/issues/53777[#53777].
```

Suggesting to remove the second paragraph - "full production support and enhanced reliability" are assumed to be part of GA anyway.

Author: Thanks for reviewing @snasovich, updated section in this [commit](https://github.com/rancher/rancher-product-docs/pull/1263/commits/3f4e0cfdda33d08ded1efd51ae87669409c0139b).

=== Behavior Changes

* By default, the `Standard User` and `Create Clusters` {rn-global-permissions}[global permissions] now allow creating and then updating CAPI infrastructure objects in the `fleet-default` namespace. Note that manually created infrastructure provider identity objects, such as `AWSClusterStaticIdentity` or `VSphereClusterIdentity`, that target the `fleet-default` namespace can be used through CAPI infrastructure objects such as `AWSCluster` or `VsphereCluster`. See https://github.com/rancher/rancher/issues/53777[#53777].
* When using native CAPI infrastructure providers with v2prov, Rancher will now generate the bootstrap cloud-config in the https://docs.cloud-init.io/en/latest/explanation/format/jinja.html[Jinja format]. User-data for regular node drivers with v2prov is unaffected. See https://github.com/rancher/rancher/issues/53777[#53777].

> [#_known_issues]

Reviewer 3: ```suggestion
[#_known_issues_k3s_and_rke2_provisioning]
```

=== Known Issues

* A known issue exists related to CAPI infrastructure providers within Rancher’s V2 provisioning framework. Creating or editing v2prov clusters with native infrastructure providers (such as CAPA or CAPV) from the UI can under some failure conditions leave unused, orphaned infrastructure resources (e.g., `AWSMachineTemplate` or `AWSCluster`) in the management cluster which need to be cleaned up manually. See https://github.com/rancher/rancher/issues/55752[#55752].

[#_continuous_delivery_fleet]
== Continuous Delivery (Fleet)

> [#_behavior_changes_3]

Reviewer 3: ```suggestion
[#_behavior_changes_continuous_delivery_fleet]
```

=== Behavior Changes

* **Deprecation**: Fleet will be deprecating support for `GitRepoRestriction` in a future release, making the use of the Policy resource for configuring multi-tenant environments the only option. See https://github.com/rancher/rancher/issues/55652[#55652].

> * **Deprecation**: Fleet support for unauthenticated webhook calls will be deprecated in a future release, making the setup of a {rn-fleet-webhook}[webhook secret] the only option. See https://github.com/rancher/rancher/issues/55633[#55633].

Reviewer 3: > deprecated in a future release

Are 55652 and 55633 intended to be the deprecation announcements? i.e. both of these deprecation entries are intended to say `removed` in a future release instead?

With the current phrasing it sounds like we're making an announcement of an announcement.

Author: Thanks for catching Billy, didn't intend to say deprecated twice essentially fixed in this [commit](https://github.com/rancher/rancher-product-docs/pull/1263/commits/29f2a7478588a84eb11650e41a39b04d1e6f0524).

[#_role_based_access_control_rbac]
== Role-Based Access Control (RBAC)

> [#_features_and_enhancements_5]

Reviewer 3: ```suggestion
[#_features_and_enhancements_role_based_access_control_rbac]
```

=== Features and Enhancements

> * Rancher introduces a new `inheritedNamespacedRules` field for GlobalRoles, allowing administrators to automatically grant users RBAC permissions to specific namespaces across all downstream clusters (for example, shared monitoring or logging namespaces). Rules defined in `inheritedNamespacedRules` automatically propagate to all downstream clusters managed by Rancher. If the namespace doesn't exist on a cluster no rules are added, however, if the namespace does not exist but gets created, the controller triggers and creates the rules. Please note that these inherited rules apply exclusively to downstream clusters and do not affect the local management cluster.

Reviewer 2: ```suggestion
* Rancher introduces a new `inheritedNamespacedRules` field for GlobalRoles, allowing administrators to automatically grant users RBAC permissions to specific namespaces across all downstream clusters (for example, shared monitoring or logging namespaces). Rules defined in `inheritedNamespacedRules` automatically propagate to all downstream clusters managed by Rancher. If the namespace doesn't exist on a cluster no rules are added. However, if the namespace does not exist but gets created, the controller triggers and creates the rules. Please note that these inherited rules apply exclusively to downstream clusters and do not affect the local management cluster.
```

** Configuring `inheritedNamespacedRules` currently requires editing the GlobalRole YAML directly or via the API, as direct UI inputs for this specific map are not yet available. Example YAML configuration for a GlobalRole with `inheritedNamespacedRules`:
+
[source,yaml]
----
apiVersion: management.cattle.io/v3
kind: GlobalRole
metadata:
namespace: default
name: test-gr-1
rules:
- apiGroups:
- catalog.cattle.io
resources:
- apps
verbs:
- create
inheritedNamespacedRules:
shared-namespace:
    - apiGroups:
    - catalog.cattle.io
    resources:
    - apps
    verbs:
    - edit
----
+
See https://github.com/rancher/rancher/issues/50663[#50663] for more information.

[#_virtualization_management_harvester]
== Virtualization Management (Harvester)

> [#_features_and_enhancements_6]

Reviewer 3: ```suggestion
[#_features_and_enhancements_virtualization_management_harvester]
```

[#_versions]
== Versions

[#_images]
=== Images

* rancher/rancher:{release-version}

[#_tools]
=== Tools

* CLI - https://github.com/rancher/cli/releases/tag/{release-version}[{release-version}]

[#_kubernetes_versions_for_rke2_k3s]
=== Kubernetes Versions for RKE2/K3s

* v1.36.1 (Default)
* v1.35.6
* v1.34.9

=== Rancher Helm Chart Versions

In Rancher v2.6.0 and later, in the *Apps & Marketplace* UI, many Rancher Helm charts are named with a major version that starts with _100_. This avoids simultaneous upstream changes and Rancher changes from causing conflicting version increments. This also complies with semantic versioning (SemVer), which is a requirement for Helm. You can see the upstream version number of a chart in the build metadata, for example: `100.0.0+up2.1.0`. See https://github.com/rancher/rancher/issues/32294[#32294].

[#_future_rancher_behavior_changes]
== Future Rancher Behavior Changes

[#_retention_policy_for_rancher_app_charts]

> === Retention Policy for Rancher App Charts

Reviewer 3: This is placed under "future changes", but the details say this is coming into effect with 2.15. Given that, shouldn't this be consolidated with all the 2.15.0 information at the start of the document?

Author: Def agreed, moved to Rancher General behavior changes [here](https://github.com/rancher/rancher-product-docs/pull/1263/commits/213421e752a57201357474cb3c089650ff0ebb0f).

[#_behavior_changes_1]
=== Behavior Changes

> * The option to disable the feature flag `ui-sql-cache` / Vai / Server-Side Pagination (SSP) in the Rancher UI has been removed in preparation for Rancher v2.16, where the feature will be enabled by default. See https://github.com/rancher/dashboard/issues/16822[#16822].

Reviewer 3: This [change](https://github.com/rancher/rancher-product-docs/pull/1263/changes/c701b18feefbdd82d2ac03d8f20bffcd7de61c76) needs to be reverted. The first sentence says we're removing the option to turn it off while `enabled by default` just means it's initially on, but can be turned off. 

```suggestion
* The option to disable the feature flag `ui-sql-cache` / Vai / Server-Side Pagination (SSP) in the Rancher UI has been removed in preparation for Rancher v2.16, where the feature will always be enabled. See https://github.com/rancher/dashboard/issues/16822[#16822].
```

Author: Thanks for catching Billy, reverted in this [commit](https://github.com/rancher/rancher-product-docs/pull/1263/commits/3278067021a793f030ecd7376b8f0d7e41616a7f).
